src/Compiler/NodeVisitor/AddFormClassesNodeVisitor.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\Compiler\NodeVisitor;
+
+use Efabrica\PHPStanLatte\Compiler\NodeVisitor\Behavior\FormsNodeVisitorBehavior;
+use Efabrica\PHPStanLatte\Compiler\NodeVisitor\Behavior\FormsNodeVisitorInterface;
+use Efabrica\PHPStanLatte\Error\Error;
+use Efabrica\PHPStanLatte\Resolver\NameResolver\NameResolver;
+use PhpParser\Builder\Class_;
+use PhpParser\Builder\Method;
+use PhpParser\Builder\Param;
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\ArrayDimFetch;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Echo_;
+use PhpParser\Node\Stmt\If_;
+use PhpParser\Node\Stmt\Return_;
+use PhpParser\NodeVisitorAbstract;
+use PHPStan\Type\VerbosityLevel;
+
+final class AddFormClassesNodeVisitor extends NodeVisitorAbstract implements FormsNodeVisitorInterface
+{
+    use FormsNodeVisitorBehavior;
+
+    private NameResolver $nameResolver;
+
+    /** @var array<array{node: Node, field: string}> */
+    private array $errorFieldNodes = [];
+
+    public function __construct(NameResolver $nameResolver)
+    {
+        $this->nameResolver = $nameResolver;
+    }
+
+    public function beforeTraverse(array $nodes)
+    {
+        $this->resetForms();
+        $this->errorFieldNodes = [];
+        return null;
+    }
+
+    public function enterNode(Node $node): ?Node
+    {
+        if ($node instanceof Assign) {
+            if (!$node->expr instanceof Assign) {
+                return null;
+            }
+            $assign = $node->expr;
+
+            if (!$assign->expr instanceof ArrayDimFetch) {
+                return null;
+            }
+            $dimFetch = $assign->expr;
+
+            if (!$dimFetch->dim instanceof String_) {
+                return null;
+            }
+            $formNameString = $dimFetch->dim;
+            $formName = $formNameString->value;
+
+            $formClassName = $this->formClassNames[$formName] ?? null;
+            if ($formClassName === null) {
+                return null;
+            }
+            $this->actualForm = $this->forms[$formName] ?? null;
```

Comment:
I think we should add __compilerError with `Form with name *** probably does not exists` here when form is not found

Reply:
👍 good idea, just have to think what to do with rest of array fetches on non-existng form 🤔

Reply:
Maybe just use generic `\Nette\Forms\Form` instead of defined fictional class?

Reply:
I think then there will be errors like cannot call ->getControl() / ->getLabel() on mixed. Ideal scenario would be to remove whole form part and keep just error about undefined form

Reply:
I added this to features in my Forms task. Will be solved later.